### PR TITLE
oci-seccomp-bpf-hook: new at 1.2.0

### PR DESCRIPTION
**Pull request by saschagrunert**
###### Motivation for this change
Add a new package for the OCI seccomp BPF hook package.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

cc @NixOS/podman

### CODE REVIEW

File: pkgs/applications/virtualization/oci-seccomp-bpf-hook/default.nix [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    installManPage docs/*.[1-9]
+  '';
+
+  meta = with stdenv.lib; {
+    homepage = "https://github.com/containers/oci-seccomp-bpf-hook";
+    description = ''
+      OCI hook to trace syscalls and generate a seccomp profile
+    '';
+    license = licenses.asl20;
+    maintainers = with maintainers; [ ] ++ teams.podman.members;
```
Comment (zowoq) [inline pkgs/applications/virtualization/oci-seccomp-bpf-hook/default.nix:55]: I don't think this should be added to the team as it is an OCI tool and not directly related to podman/cri-o.

Comment (saschagrunert) [reply pkgs/applications/virtualization/oci-seccomp-bpf-hook/default.nix:55]: Not? I mean I can remove the team if you prefer it, but the hook would be used mostly from Podman and CRI-O from my point of view. :)

File: pkgs/applications/virtualization/oci-seccomp-bpf-hook/default.nix [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    libseccomp
+    linuxPackages.bcc
+  ];
+
+  buildPhase = ''
+    make
+  '';
+
+  postBuild = ''
+    sed -i "s;HOOK_BIN_DIR;$out/bin;g" oci-seccomp-bpf-hook.json
```
Comment (zowoq) [inline pkgs/applications/virtualization/oci-seccomp-bpf-hook/default.nix:40]: Could use `substituteInPlace` here.

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -24521,6 +24521,8 @@ in
 
   nxengine-evo = callPackage ../games/nxengine-evo { };
 
+  oci-seccomp-bpf-hook = callPackage ../applications/virtualization/oci-seccomp-bpf-hook/default.nix { };
```
Comment (zowoq) [inline pkgs/top-level/all-packages.nix:24524]:
```suggestion
  oci-seccomp-bpf-hook = callPackage ../applications/virtualization/oci-seccomp-bpf-hook { };
```

File: pkgs/applications/virtualization/oci-seccomp-bpf-hook/default.nix (hunk 1 above)
Comment (zowoq) [reply pkgs/applications/virtualization/oci-seccomp-bpf-hook/default.nix:55]: I'd prefer to remove it for now and we can add it to the team if/when it is added to the wrappers.

Comment (zowoq) [reply pkgs/applications/virtualization/oci-seccomp-bpf-hook/default.nix:55]: Sorry, I posted this comment and then deleted it by accident:

I'd prefer to remove it for now and add it if/when we add it to the wrappers.

Comment (saschagrunert) [reply pkgs/applications/virtualization/oci-seccomp-bpf-hook/default.nix:55]: Done :+1:

File: pkgs/applications/virtualization/oci-seccomp-bpf-hook/default.nix (hunk 2 above)
Comment (saschagrunert) [reply pkgs/applications/virtualization/oci-seccomp-bpf-hook/default.nix:40]: Fixed :+1:

File: pkgs/top-level/all-packages.nix (hunk 3 above)
Comment (saschagrunert) [reply pkgs/top-level/all-packages.nix:24524]: Yep, fixed :+1:

**Review by zowoq — APPROVED** [review]

**Comment by zowoq** [comment]
@ofborg build oci-seccomp-bpf-hook

**Comment by zowoq** [comment]
@saschagrunert Do you want to try fixing aarch64 build failure before we merge? If not I'm fine with just disabling the package on aarch64.

**Comment by saschagrunert** [comment]
> @saschagrunert Do you want to try fixing aarch64 build failure before we merge? If not I'm fine with just disabling the package on aarch64.

Hm yeah let's exclude it for now. I check if I can fix this in the upstream sources.

**Comment by saschagrunert** [comment]
@ofborg build oci-seccomp-bpf-hook

File: pkgs/applications/virtualization/oci-seccomp-bpf-hook/default.nix [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  pname = "oci-seccomp-bpf-hook";
+  version = "1.2.0";
+  src = fetchFromGitHub {
+    owner = "containers";
+    repo = "oci-seccomp-bpf-hook";
+    rev = "v${version}";
+    sha256 = "143x4daixzhhhpli1l14r7dr7dn3q42w8dddr16jzhhwighsirqw";
+  };
+  vendorSha256 = null;
+  badPlatforms = [ "aarch64-linux" ];
```
Comment (saschagrunert) [inline pkgs/applications/virtualization/oci-seccomp-bpf-hook/default.nix:21]: @zowoq is this enough to exclude the architecture? :innocent:

Comment (zowoq) [reply pkgs/applications/virtualization/oci-seccomp-bpf-hook/default.nix:21]: I think `badPlatforms` needs to be in `meta` with `platforms`.

Comment (saschagrunert) [reply pkgs/applications/virtualization/oci-seccomp-bpf-hook/default.nix]: Hah, thank you for the hint! This one was new to me :)

**Comment by saschagrunert** [comment]
@ofborg build oci-seccomp-bpf-hook

**Pull request merged**